lib/scanner/walk.go

```diff
@@ -476,8 +479,11 @@ func SymlinkTypeEqual(disk, index uint32) bool {
 	// know means you are on Unix, and on Unix you don't really care what the
 	// target type is. The moment you do know, and if something doesn't match,
 	// that will propagate through the cluster.
-	if disk&protocol.FlagSymlinkMissingTarget != 0 && index&protocol.FlagSymlinkMissingTarget == 0 {
+	if disk == symlinks.TargetUnknown {
 		return true
 	}
-	return disk&protocol.SymlinkTypeMask == index&protocol.SymlinkTypeMask
+	if f.IsDirectory() {
+		return disk == symlinks.TargetDirectory
+	}
+	return disk == symlinks.TargetFile
```

Comment:
It depends on what you'd expect to happen. I prefer windows nodes follow type changes, even if the target is unknwon and the unknown bit is set. For example 3 windows nodes. All have a file link with no valid target ( which defaults to a file type link, ), node 1 goes away, node 2 suddently finds a target which is dir, both node 2 and 3 change link type to dir. Now the target vanishes again, and node 1 comes back online. They have different type of links, but are still in sync, as the equality holds. Obviously the problem goes away when a target is found, but I feel its inconsistent for the time being.